### PR TITLE
Ctm/random releases

### PR DESCRIPTION
**Pull request by canadathemusical**
image

### CODE REVIEW

File: rust/src/collection.rs [hunk 1]
```diff
@@ -78,3 +91,76 @@ pub struct Label {
     pub id: i64,
     pub resource_url: String,
 }
+
+fn clean_artist_name(artist: String) -> String {
+    let regex = Regex::new(r"\s*\(\d+\)\s*").unwrap();
```
Comment (resnickmicah) [inline rust/src/collection.rs:96]: This is just removing whitespace, right? https://doc.rust-lang.org/std/primitive.str.html#method.trim you won't need regex for that.

File: rust/src/collection.rs [hunk 2]
```diff
@@ -78,3 +91,76 @@ pub struct Label {
...
+
+    format!("{} - {}", artist, title)
+}
+
+fn get_random_releases(releases: Vec<Album>) -> Vec<Album> {
+    let mut random_releases: Vec<Album> = Vec::new();
+    for _ in 0..4 {
+        let random_index = rand::thread_rng().gen_range(0..releases.len());
+        random_releases.push(releases[random_index].clone());
+    }
```
Comment (resnickmicah) [inline rust/src/collection.rs]: The odds of this happening are pretty low, but if you want to ensure four distinct random indices (without repeats) you can use `choose_multiple`: https://docs.rs/rand/latest/rand/seq/trait.SliceRandom.html#tymethod.choose_multiple

https://chatgpt.com/share/6702eb23-4d58-8006-869d-06e829cbb4b4

**Review by resnickmicah — DISMISSED** [review]
Not blocking merge, but just pointing out a few things.

File: rust/src/collection.rs (hunk 1 above)
Comment (canadathemusical) [reply rust/src/collection.rs:96]: that is not the case. some artists share names or have multiple artist entries. for instance "Electric Wizard (2)". This removed the " (2)" in such a case

File: rust/src/collection.rs (hunk 2 above)
Comment (canadathemusical) [reply rust/src/collection.rs]: done

**Review by resnickmicah — APPROVED** [review]

**Pull request merged**